Build fails on Windows (mingw64) using ghcup

GHC Version 9.6.6
Cabal Version 3.10.3.0

affected: bytestring-0.12.2.0
**not** affected: bytestring-0.11.5.4

on mingw64 Windows build

```bash
[31 of 31] Compiling Data.ByteString.Builder ( Data\ByteString\Builder.hs, dist\build\Data\ByteString\Builder.o )
In file included from cbits\fpstring.c:32:0: error: 
In file included from C:\Tools\ghc-9.6.5\lib\x86_64-windows-ghc-9.6.5\rts-1.0.2\include\HsFFI.h:24:0: error:
    
In file included from C:\Tools\ghc-9.6.5\lib\x86_64-windows-ghc-9.6.5\rts-1.0.2\include\stg\Types.h:32:0: error:
    
In file included from C:\msys64\mingw64\include\inttypes.h:11:0: error:
    
In file included from C:\msys64\mingw64\include\crtdefs.h:10:0: error:
    
In file included from C:\msys64\mingw64\include\corecrt.h:10:0: error:
    
In file included from C:\msys64\mingw64\include\_mingw.h:10:0: error:
    
C:\msys64\mingw64\include\_mingw_mac.h:328:29: error:
     error: '_FORTIFY_SOURCE' is not defined, evaluates to 0 [-Werror,-Wundef]
    |
328 | #if defined(__MINGW32__) && _FORTIFY_SOURCE > 0 && __OPTIMIZE__ > 0 \
    |                             ^
#if defined(__MINGW32__) && _FORTIFY_SOURCE > 0 && __OPTIMIZE__ > 0 \
                            ^
1 error generated.
`clang.exe' failed in phase `C Compiler'. (Exit code: 1)
Error: cabal.exe: Failed to build bytestring-0.12.2.0 (which is required by
hsqml-0.3.6.0). See the build log above for details.
```

--- a/bytestring.cabal
+++ b/bytestring.cabal
@@ -150,7 +150,7 @@ library
                    Data.ByteString.Utils.UnalignedAccess
 
   ghc-options:     -Wall -fwarn-tabs -Wincomplete-uni-patterns
-                   -optP-Wall -optP-Werror=undef
+                   -optP-Wall -optP-Wundef
                    -O2
                    -fmax-simplifier-iterations=10
                    -fdicts-cheap
@@ -178,7 +178,7 @@ library
     -- DNDEBUG disables asserts in cbits/
     cc-options:        -std=c11 -DNDEBUG=1
                        -fno-strict-aliasing
-                       -Werror=undef
+                       -Wundef
 
     -- No need to link to libgcc on ghc-9.4 and later which uses a clang-based
     -- toolchain.

